java/code/src/com/redhat/rhn/taskomatic/task/HubReportDbUpdateWorker.java

```diff
@@ -157,10 +157,16 @@ public void run() {
             }
         }
         catch (Exception e) {
+            e.printStackTrace();
+            parentQueue.getQueueRun().failed();
+            parentQueue.changeRun(null);
+            parentQueue.workerDone();
+            HibernateFactory.closeSession();
```

Comment:
These two instructions are also in the `finally` block, they will be executed twice in case of exception.